src/Policies/MenuItemPolicy.php

```diff
@@ -43,10 +43,10 @@ protected function checkPermission(User $user, $model, $action)
         }
 
         // If permission doesn't exist, we can't check it!
-        if (!self::$permissions->contains('key', 'browse_'.$slug)) {
+        if (!self::$permissions->contains('key', $action.'_'.$slug)) {
```

Comment:
As mentioned by @emptynick, we need to guarantee `$action` is never null or empty string.  Can you add a check above this similar to:
```
if ($action == null || empty($action)) {
    $action = 'browse';
}
```

Reply:
- First of all `empty ($action)` function already includes check for `null` value.
- Secondly, with all due respect, I disagree with the necessity of this check. We don't do the similar checks in other policies, why we should do it here?

Reply:
Because `$action` was always `browse` (because it was hardcoded)
So someone could use `checkPermission($user, $model, '')` and it would work.
If we don't check it, it wont work anymore so its kind of a breaking change

Reply:
Exactly as @emptynick said.  Just because others don't do it doesn't mean this shouldn't.  If we change other policies to this dynamic approach, then they will need a default set as well.  Thanks for making the quick change.